backend/src/v2/component/launcher_v2_test.go

```diff
@@ -237,3 +240,197 @@ func Test_get_log_Writer(t *testing.T) {
+			if test.expectedErr != nil {
+				assert.ErrorContains(t, err, test.expectedErr.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func Test_DefaultLauncherV2Dependencies(t *testing.T) {
+	getDefaultK8sClientOrig := getDefaultK8sClient
+	getDefaultMetadataClientOrig := getDefaultMetadataClient
+	getDefaultCacheClientOrig := getDefaultCacheClient
+	defer func() {
+		getDefaultK8sClient = getDefaultK8sClientOrig
+		getDefaultMetadataClient = getDefaultMetadataClientOrig
+		getDefaultCacheClient = getDefaultCacheClientOrig
+	}()
+
+	type clientProviders struct {
+		getDefaultK8sClient      func() (kubernetes.Interface, error)
+		getDefaultMetadataClient func(address string, port string) (metadata.ClientInterface, error)
+		getDefaultCacheClient    func() (cacheutils.ClientInterface, error)
+	}
+	happyClientProviders := clientProviders{
+		getDefaultK8sClient: func() (kubernetes.Interface, error) {
+			return fake.NewSimpleClientset(), nil
+		},
+		getDefaultMetadataClient: func(address string, port string) (metadata.ClientInterface, error) {
+			return &metadata.Client{}, nil
+		},
+		getDefaultCacheClient: func() (cacheutils.ClientInterface, error) {
+			return &cacheutils.Client{}, nil
+		},
+	}
+
+	tests := []struct {
+		name        string
+		providers   clientProviders
+		expectedErr error
+	}{
+		{
+			name:        "happy path",
+			providers:   happyClientProviders,
+			expectedErr: nil,
+		},
+		{
+			name: "kubernetes client error",
+			providers: func() clientProviders {
+				myClientProviders := happyClientProviders
+				myClientProviders.getDefaultK8sClient = func() (kubernetes.Interface, error) {
+					return nil, errors.New("error getting kubernetes client")
+				}
+				return myClientProviders
+			}(),
+			expectedErr: errors.New("error getting kubernetes client"),
+		},
+		{
+			name: "metadata client error",
+			providers: func() clientProviders {
+				myClientProviders := happyClientProviders
+				myClientProviders.getDefaultMetadataClient = func(address string, port string) (metadata.ClientInterface, error) {
+					return nil, errors.New("error getting metadata client")
+				}
+				return myClientProviders
+			}(),
+			expectedErr: errors.New("error getting metadata client"),
+		},
+		{
+			name: "cache client error",
+			providers: func() clientProviders {
+				myClientProviders := happyClientProviders
+				myClientProviders.getDefaultCacheClient = func() (cacheutils.ClientInterface, error) {
+					return nil, errors.New("error getting cache client")
+				}
+				return myClientProviders
+			}(),
+			expectedErr: errors.New("error getting cache client"),
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			getDefaultK8sClient = test.providers.getDefaultK8sClient
+			getDefaultMetadataClient = test.providers.getDefaultMetadataClient
+			getDefaultCacheClient = test.providers.getDefaultCacheClient
+
+			deps, err := DefaultLauncherV2Dependencies("test-address", "test-port")
```

Comment:
tbh I don't see much value in this test as it is, open to being persuaded

atm it seems like  `DefaultLauncherV2Dependencies` is a simple function that calls the mocked functions, and we're just testing that yes indeed if a function returns an error, it will return an error 🤷🏾 (I don't think we need to confirm that works in golang). 

I think it would be more interesting if we tested malformed configs raised client errors when trying to create a launcher `NewLauncherV2()` but off the top off my head that may not be trivial to reproduce (you can give it a shot though if you want!)